core/tee/tee_svc_storage.c

```diff
@@ -626,6 +627,9 @@ TEE_Result syscall_storage_obj_create(unsigned long storage_id, void *object_id,
 		goto err;
 	}
 
+	/* remove the file if it exists*/
+	if (!filedoesnotexist)
+		tee_file_ops.unlink(file);
```

Comment:
Current implementation is really ad-hoc. This is a reimplementation of posix-like instruction, but do not follow exactly the spec. So I do not think the current open will solve it.

With the new implementation, I understand that "rename" will really fit the posix specification, that is will delete the file first under right circumstances, so I think that with the new implementation, the "unlink" I added line 632 is useless.

@jenswi-linaro If I'm right, should I only add a comment "Fixme: unlink must be removed once SQLite is used", and update the commit message indicating the create operation is not atomic in case the file exists (prefered way)? or should I wait till SQLite implementation is in place?